plugins/in_syslog/syslog_prot.c

```diff
@@ -31,88 +33,126 @@ static inline void consume_bytes(char *buf, int bytes, int length)
-                                             char *base_object_buffer,
-                                             size_t base_object_size,
-                                             char *raw_message_buffer,
-                                             size_t raw_message_size)
+static int append_message_to_record_data(char **result_buffer,
+                                         size_t *result_size,
+                                         flb_sds_t message_key_name,
+                                         char *base_object_buffer,
+                                         size_t base_object_size,
+                                         char *message_buffer,
+                                         size_t message_size,
+                                         int message_type)
 {
-    int                i;
     int                result;
-    size_t             unpacker_offset;
-    msgpack_sbuffer    mp_sbuf;
-    msgpack_packer     mp_pck;
-    msgpack_unpacked   unpacked_buffer;
+    char              *modified_data_buffer;
+    int                modified_data_size;
+    msgpack_object_kv *new_map_entries[1];
+    msgpack_object_kv  message_entry;
     *result_buffer = NULL;
     *result_size = 0;
+    modified_data_buffer = NULL;
 
-    unpacker_offset = 0;
-    msgpack_unpacked_init(&unpacked_buffer);
-    result = msgpack_unpack_next(&unpacked_buffer,
-                                 base_object_buffer,
-                                 base_object_size,
-                                 &unpacker_offset);
-
-    if (result != MSGPACK_UNPACK_SUCCESS) {
-        return -1;
-    }
-
-    if (unpacker_offset != base_object_size) {
-        msgpack_unpacked_destroy(&unpacked_buffer);
-        return -2;
-    }
+    if (message_key_name != NULL) {
+        new_map_entries[0] = &message_entry;
 
-    if (unpacked_buffer.data.type != MSGPACK_OBJECT_MAP) {
-        msgpack_unpacked_destroy(&unpacked_buffer);
-        return -3;
-    }
+        message_entry.key.type = MSGPACK_OBJECT_STR;
+        message_entry.key.via.str.size = flb_sds_len(message_key_name);
+        message_entry.key.via.str.ptr  = message_key_name;
 
-    msgpack_sbuffer_init(&mp_sbuf);
-    msgpack_packer_init(&mp_pck, &mp_sbuf, msgpack_sbuffer_write);
+        if (message_type == MSGPACK_OBJECT_BIN) {
+            message_entry.val.type = MSGPACK_OBJECT_BIN;
+            message_entry.val.via.bin.size = message_size;
+            message_entry.val.via.bin.ptr  = message_buffer;
+        }
+        else if (message_type == MSGPACK_OBJECT_STR) {
+            message_entry.val.type = MSGPACK_OBJECT_STR;
+            message_entry.val.via.str.size = message_size;
+            message_entry.val.via.str.ptr  = message_buffer;
+        }
+        else {
+            result = FLB_MAP_NOT_MODIFIED;
 
-    msgpack_pack_map(&mp_pck, unpacked_buffer.data.via.map.size + 1);
+            return result;
```

Comment:
Remove this return and wrap the code in lines 77-84 in a conditional that only executes if `result` equals `FLB_MAP_NOT_MODIFIED`